tests/ui/transmutability/enums/repr/should_require_well_defined_layout.stderr

```diff
@@ -2,7 +2,10 @@ error[E0277]: `void::repr_rust` cannot be safely transmuted into `()` in the def
   --> $DIR/should_require_well_defined_layout.rs:28:52
    |
 LL |         assert::is_maybe_transmutable::<repr_rust, ()>();
-   |                                                    ^^ `void::repr_rust` cannot be safely transmuted into `()` in the defining scope of `assert::Context`.
+   |                                                    ^^
+   |                                                    |
+   |                                                    `void::repr_rust` cannot be safely transmuted into `()` in the defining scope of `assert::Context`.
+   |                                                    Reason: SrcIsUnspecified
```

Comment:
Instead of:
```
`void::repr_rust` cannot be safely transmuted into `()` in the defining scope of `assert::Context`.
Reason: SrcIsUnspecified
```
I'd suggest:
```
`void::repr_rust` cannot be safely transmuted into `()` in the defining scope of `assert::Context` because `void::repr_rust` does not have a well-specified layout.
```

Reply:
Is it possible to combine them somehow? The first line comes from the on_unimplemented macro and the other line comes from the actual error.

Reply:
It's probably better to just remove the `rustc_on_unimplemented` from the macro and handle formatting the error message for the transmute trait manually.